src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Confirm.tsx

```diff
@@ -0,0 +1,194 @@
+import { WithdrawContext } from '../WithdrawContext'
+
+export const Confirm = ({ onNext }: StepComponentProps) => {
+  const [gasLimit, setGasLimit] = useState<string | null>(null)
+  const [gasPrice, setGasPrice] = useState<string | null>(null)
+  const { state, dispatch } = useContext(WithdrawContext)
+  const translate = useTranslate()
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+
+  const assetNamespace = 'slip44' // TODO: add to query, why do we hardcode this?
+  // Asset info
+  const underlyingAssetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+  const underlyingAsset = useAppSelector(state => selectAssetById(state, underlyingAssetId))
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+
+  const unbondingDays = useMemo(() => assetIdToUnbondingDays(assetId), [assetId])
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  // user info
+  const { state: walletState } = useWallet()
+
+  const { handleStakingAction } = useStakingAction()
+
+  useEffect(() => {
+    ;(async () => {
+      const { gasLimit, gasPrice } = await getStakingFees(asset, feeMarketData.price)
+
+      setGasLimit(gasLimit)
+      setGasPrice(gasPrice)
+    })()
+  }, [asset, asset.precision, feeMarketData.price])
+
+  const feeAssetBalance = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByAssetId(state, { assetId: feeAsset?.assetId ?? '' }),
+  )
+
+  if (!state || !dispatch) return null
+
+  const handleConfirm = async () => {
+    try {
+      if (!state.userAddress || !walletState.wallet || state.loading || !gasLimit || !gasPrice)
+        return
+      dispatch({ type: CosmosWithdrawActionType.SET_LOADING, payload: true })
+
+      const broadcastTxId = await handleStakingAction({
+        asset,
+        validator: contractAddress,
+        chainSpecific: {
+          gas: gasLimit,
+          fee: bnOrZero(gasPrice).times(`1e+${asset?.precision}`).toString(),
+        },
+        value: bnOrZero(state.withdraw.cryptoAmount).times(`1e+${asset.precision}`).toString(),
+        action: StakingAction.Unstake,
+      })
+
+      dispatch({
+        type: CosmosWithdrawActionType.SET_WITHDRAW,
+        payload: {
+          txStatus: broadcastTxId?.length ? 'success' : 'failed',
+        },
+      })
+
+      if (!broadcastTxId) {
+        throw new Error() // TODO
+      }
+
+      dispatch({ type: CosmosWithdrawActionType.SET_TXID, payload: broadcastTxId })
+    } catch (error) {
+      console.error('CosmosWithdraw:handleConfirm error', error)
```

Comment:
moduleLogger

Reply:
Also from the Foxy implementation, will add it in the same stacked PR if that's fine by you

Reply:
https://github.com/shapeshift/web/pull/2323